src/stack/Stack.java

```diff
@@ -19,38 +21,73 @@ public Stack(int size) {
 	 * Add element into stack
 	 */
 	public void push(int value) {
-
+		if (tos == data.length-1) {
+			ensureCapacity();
+			data[++tos] = value;
```

Comment:
You don't need **data[++tos] = value;** inside **IF statement**
and also, remove **else statement**
and put **data[++tos] = value;** after **IF statement**